File: lab_4/Circle.cpp
#include "Circle.hpp"

Circle::Circle(double a) {
    this->a = a;
    std::cout << "Circle constructor" << std::endl;
}

Circle::~Circle() {
    std::cout << "Circle destructor" << std::endl;
}

double Circle::countLength() {
    return 2 * M_PI * this->a;
}

double Circle::countSquare() {
    return pow(this->a, 2) * M_PI;
}

void Circle::view() {
    std::cout << "Circle radius = " << this->a << std::endl;
}

File: lab4/src/Circle/Circle.java
package Circle;

public class Circle {
    protected double a;

    public Circle(double a) {
        this.a = a;
        System.out.println("Circle constructor with parametrs");
    }
    public Circle() {
        this(1);
        System.out.println("Circle constructor default");
    }

    public double countLength() {
        return 2 * Math.PI * a;
    }

    public double countSquare() {
        return Math.PI *Math.pow(a, 2);
    }

    public void view() {
        System.out.println("Circle radius = " + this.a);
    }
}
